service/storage_proxy.cc

```diff
@@ -5481,14 +5481,23 @@ storage_proxy::query_partition_key_range_concurrent(storage_proxy::clock_type::t
                 break;
             } else if (pcf) {
                 // check that merged set hit rate is not to low
-                auto find_min = [&g = _remote->gossiper(), pcf] (const inet_address_vector_replica_set& range) {
+                auto find_min = [this, pcf] (const inet_address_vector_replica_set& range) {
+                    if (range.size() == 1 && range[0] == utils::fb_utilities::get_broadcast_address()) {
```

Comment:
The `inet_addresds_vector_replica_set()` seem to deserve having `bool only_me() const` method to be used here and in previous patch